wolfssl/src/ssl.rs

```diff
@@ -1100,6 +1111,95 @@ impl<IOCB: IOCallbacks> Session<IOCB> {
             e => unreachable!("{e:?}"),
         }
     }
+
+    /// Enable key logging for applications
+    #[cfg(feature = "debug")]
+    pub fn enable_key_logging(&mut self, keylog: Arc<dyn Tls13SecretCallbacks>) -> Result<()> {
+        self.keylog = Some(keylog.clone());
+
+        let keylog_c = Box::into_raw(Box::new(keylog));
+
+        // SAFETY: No documentation available for [`wolfSSL_KeepArrays`] [`wolfSSL_set_tls13_secret_cb`].
+        // But based on api implementation, it expects a valid pointer to `WOLFSSL`.
```

Comment:
I think we could use some discussion of the safety aspects relating to the lifetime of `keylog_c` and the memory behind the keylog `Arc`.

Do you need the box here, I think [`Arc::as_ptr`](https://doc.rust-lang.org/std/sync/struct.Arc.html#method.as_ptr) gives you the raw pointer you need and the docs say:
> The pointer is valid for as long as there are strong counts in the Arc.

I think in the cb you could use `ctx as *mut Arc<dyn Tls13SecretCallbacks>` as a ptr without needing the `Box`? Maybe? For IOCB we use `let io = unsafe { &*(ctx as *mut IOCB) };` to recover it.